workers/data_refinery_workers/processors/create_compendia.py

```diff
@@ -0,0 +1,218 @@
+
+    Via https://github.com/AlexsLemonade/refinebio/issues/508#issuecomment-435879283: 
+     - Combine all microarray samples with a full join to form a microarray_expression_matrix (this may end up being a DataFrame)
+     - Combine all RNA-seq samples (lengthScaledTPM) with a full outer join to form a rnaseq_expression_matrix
+     - Calculate the sum of the lengthScaledTPM values for each row (gene) of the rnaseq_expression_matrix (rnaseq_row_sums)
+     - Calculate the 10th percentile of rnaseq_row_sums
+     - Drop all rows in rnaseq_expression_matrix with a row sum < 10th percentile of rnaseq_row_sums; this is now filtered_rnaseq_matrix
+     - log2(x + 1) transform filtered_rnaseq_matrix; this is now log2_rnaseq_matrix
+     - Set all zero values in log2_rnaseq_matrix to NA, but make sure to keep track of where these zeroes are
+     - Perform a full outer join of microarray_expression_matrix and log2_rnaseq_matrix; combined_matrix
+     - Remove genes (rows) with >30% missing values in combined_matrix
+     - Remove samples (columns) with >50% missing values in combined_matrix
+     - "Reset" zero values that were set to NA in RNA-seq samples (i.e., make these zero again) in combined_matrix
+     - Transpose combined_matrix; transposed_matrix
+     - Perform imputation of missing values with IterativeSVD (rank=10) on the transposed_matrix; imputed_matrix
+     - Untranspose imputed_matrix (genes are now rows, samples are now columns)
+     - Quantile normalize imputed_matrix where genes are rows and samples are columns
+
+    """
+    job_context['time_start'] = timezone.now()
+
+    # Combine all microarray samples with a full join to form a microarray_expression_matrix (this may end up being a DataFrame)
+    microarray_expression_matrix = job_context['microarray_inputs']
+
+    # Combine all RNA-seq samples (lengthScaledTPM) with a full outer join to form a rnaseq_expression_matrix
+    rnaseq_expression_matrix = job_context['rnaseq_inputs']
+
+    # Calculate the sum of the lengthScaledTPM values for each row (gene) of the rnaseq_expression_matrix (rnaseq_row_sums)
+    rnaseq_row_sums = np.sum(rnaseq_expression_matrix, axis=1)
+
+    # Calculate the 10th percentile of rnaseq_row_sums
+    rnaseq_tenth_percentile = np.percentile(rnaseq_row_sums, 10)
+
+    # Drop all rows in rnaseq_expression_matrix with a row sum < 10th percentile of rnaseq_row_sums; this is now filtered_rnaseq_matrix
+    # TODO: This is probably a better way to do this with `np.where`
+    rows_to_filter = []
+    for (x, sum_val) in rnaseq_row_sums.items():
+        if sum_val < rnaseq_tenth_percentile:
+            rows_to_filter.append(x)
+
+    filtered_rnaseq_matrix = rnaseq_expression_matrix.drop(rows_to_filter)
+
+    # log2(x + 1) transform filtered_rnaseq_matrix; this is now log2_rnaseq_matrix
+    filtered_rnaseq_matrix_plus_one = filtered_rnaseq_matrix + 1
+    log2_rnaseq_matrix = np.log2(filtered_rnaseq_matrix)
```

Comment:
Should this be `np.log2(filtered_rnaseq_matrix_plus_one)` or is there something OOP-y I'm missing? Based on the fact that you need to remove `-Inf` and `Inf` values here: https://github.com/AlexsLemonade/refinebio/pull/888/files#diff-5b6c34e495c6bb6ddf4320b09359c528R141, I'm guessing the add 1 step was not successful.